### PR TITLE
[Devtools Week] Update contributing guide

### PR DESCRIPTION
**Pull request by thepetk**
## What does this PR do?

This PR updates the `contributing.md` in order to be more consistent with the rest of the `contributing.md` files inside the `Devfile` organization. The updated document follows the `contributing.md` file of the `registry-support` repo ([here](https://github.com/devfile/registry-support/blob/main/CONTRIBUTING.md))

### Which issue(s) does this PR fix

fixes https://github.com/devfile/api/issues/1351

### PR acceptance criteria

Testing and documentation do not need to be complete in order for this PR to be approved. We just need to ensure tracking issues are opened.

- [x] Unit/Functional tests
- [x] Documentation

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,33 +1,63 @@
...
+
+By default for each change in the PR, GitHub Actions and OpenShift CI will run checks against your changes (linting, unit testing, and integration tests).
+
+If comments have been given in a review, they have to be addressed before merging.
+
+After addressing review comments, don't forget to add a comment in the PR afterward, so everyone gets notified by Github and know to re-review.
+
+# Contact us
+
+If you have questions, please visit us on `#devfile` in the [Kubernetes Slack](https://slack.k8s.io).
```
Comment (michael-valdron) [inline CONTRIBUTING.md:63]:
```suggestion
If you have any questions, please visit us the `#devfile` channel under the [Kubernetes Slack](https://slack.k8s.io) workspace.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,33 +1,63 @@
...
-Origin (DCO). This document was created by the Linux Kernel community and is a
-simple statement that you, as a contributor, have the legal right to make the
-contribution. See the [DCO](DCO) file for details.
+Owners of the repository will watch out for and review new PRs.
+
+By default for each change in the PR, GitHub Actions and OpenShift CI will run checks against your changes (linting, unit testing, and integration tests).
+
+If comments have been given in a review, they have to be addressed before merging.
+
+After addressing review comments, don't forget to add a comment in the PR afterward, so everyone gets notified by Github and know to re-review.
```
Comment (michael-valdron) [inline CONTRIBUTING.md:59]: @thepetk I'm thinking the author should mention the reviewer in their comment so the reviewer can provide a follow up review, wdyt?

```suggestion
After addressing review comments, don't forget to add a comment in the PR with the reviewer mentioned afterward, so they get notified by Github to provide a re-review.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,33 +1,63 @@
...
 
-Read more about [commit types](https://github.com/pvdlg/conventional-commit-types#commit-types).
-### Certificate of Origin
+You can easily add this footer to your commits by adding `-s` when running `git commit`. When you think the code is ready for review, create a pull request and link the issue associated with it.
 
-By contributing to this project you agree to the Developer Certificate of
-Origin (DCO). This document was created by the Linux Kernel community and is a
-simple statement that you, as a contributor, have the legal right to make the
-contribution. See the [DCO](DCO) file for details.
+Owners of the repository will watch out for and review new PRs.
```
Comment (michael-valdron) [inline CONTRIBUTING.md:53]:
```suggestion
Owners of the repository will watch out for new PRs and provide reviews to them.
```

**Review by michael-valdron — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,33 +1,63 @@
...
+
+By default for each change in the PR, GitHub Actions and OpenShift CI will run checks against your changes (linting, unit testing, and integration tests).
```
[the remark this answers is not in the document]
Comment (thepetk) [reply CONTRIBUTING.md]: @michael-valdron I've removed the openshift CI as is doesn't exist as a check and added the code coverage checks.

**Comment by thepetk** [comment]
@michael-valdron I've addressed your suggestions :) thanks for the review

**Review by michael-valdron — APPROVED** [review]
/lgtm

**Pull request merged**